### ISSUE TITLE
import! does not return Result object

### ISSUE THREAD

**Issue by arinchoi03** [zdennis/activerecord-import#632]
I was trying to use something like:

```
begin
   result = klass.import! table_content, timestamps: false, batch_size: 1000
catch StandardError => e
   puts result.failed_instances
end
```

But looks like the validation error is thrown before a result object is created, so I do not have access to `failed_instances` since `result` is undefined. I got around this by not using `import!` but `import`, which _does_ return `result` and `result.failed_instances`.

Can you add in the documentation that using `import!` will never be able to return `failed_instances`?

**Comment by jkowens** [comment]
I don't believe that is entirely true. Doesn't `import!` return a `Result` object on a successful import?

**Comment by arinchoi03** [comment]
@jkowens ah, you're right, I mean when the `import!` fails!

**Comment by jkowens** [comment]
Raising an error on failure is a fairly common idiom for AR bang methods, but documentation for the `import!` method could be improved in the Readme. If someone is able to do that feel free to create a PR.
